src/AudioFormat.hxx

```diff
@@ -65,6 +65,12 @@ struct AudioFormat {
 	 */
 	uint8_t channels;
 
+	/**
+	 * Bits per sample.
+	 * If raw_bits = 0, use sample_format_size().
+	 */
+	uint8_t raw_bits = 0;
```

Comment:
Why? The commit message doesn't explain. This change adds some complexity which must be justified.

Reply:
Just for caryaudio. We need to known the real bits in the sample. But sample_format_size()  report the same size, which is 32, for SampleFormat::S24_P32 and SampleFormat::S32. And when playing 64 bits files, it's still report 32 bits.